app/src/androidTest/java/com/github/wnder/InternalCachePictureDatabaseOfflineTest.java

```diff
@@ -0,0 +1,202 @@
+        scoreboard = new HashMap<>();
+        scoreboard.put("testUser", 200.);
+        iDirectory = context.getDir("images", Context.MODE_PRIVATE);
+        mDirectory = context.getDir("metadata", Context.MODE_PRIVATE);
+
+        int currILength = iDirectory.listFiles().length;
+        int currDLength = mDirectory.listFiles().length;
+
+        picture = new LocalPicture(uniqueId, bmp, realLoc, guessLoc, scoreboard);
+
+        ICPD.storePictureLocally(picture);
+        //asserts ensuring both files were correctly created
+        assertThat(iDirectory.listFiles().length, is(currILength + 1));
+        assertThat(mDirectory.listFiles().length, is(currDLength + 1));
+    }
+
+    @AfterClass
+    public static void teardown(){
+        int currILength = iDirectory.listFiles().length;
+        int currDLength = mDirectory.listFiles().length;
+        ICPD.deleteLocalPicture(uniqueId);
+        //Asserts ensuring both files were correctly deleted
+        assertThat(iDirectory.listFiles().length, is(currILength - 1));
+        assertThat(mDirectory.listFiles().length, is(currDLength - 1));
+    }
+
+    @Test
+    public void updateAndGetScoreboardWork() throws ExecutionException, InterruptedException {
+        Map<String, Double> scoreboard = ICPD.getScoreboard(uniqueId).get();
+        assertThat(scoreboard.get("testUser"), is(200.));
+
+        scoreboard.put("testUser", 150.);
+        ICPD.updateLocalScoreboard(uniqueId, scoreboard);
+
+        Map<String, Double> newScoreboard = ICPD.getScoreboard(uniqueId).get();
+        assertThat(newScoreboard.get("testUser"), is(150.));
+    }
+
+    @Test
+    public void getLocationWorks() throws ExecutionException, InterruptedException {
+        Location storedLoc = ICPD.getLocation(uniqueId).get();
+        assertThat(storedLoc.getLongitude(), is(realLoc.getLongitude()));
+        assertThat(storedLoc.getLatitude(), is(realLoc.getLatitude()));
+    }
+
+    @Test
+    public void getGuessedLocationWorks(){
+        Location storedLoc = ICPD.getLocalGuessedLocation(uniqueId);
+        assertThat(storedLoc.getLongitude(), is(guessLoc.getLongitude()));
+        assertThat(storedLoc.getLatitude(), is(guessLoc.getLatitude()));
+    }
+
+    @Test
+    public void getPictureWorks() throws ExecutionException, InterruptedException {
+        Bitmap readFile = ICPD.getBitmap(uniqueId).get();
+        int w1 = bmp.getWidth();
+        int w2 = readFile.getWidth();
+        assertThat(w1, is(w2));
+    }
+
+    @Test
+    public void getApproximateLocationThrows(){
+        try {
+            ICPD.getApproximateLocation(uniqueId);
+            fail();
+        }
+        catch(IllegalStateException e){
+            assertTrue(true);
+        }
+    }
+
+    @Test
+    public void getUserGuessesThrows(){
+        try {
+            ICPD.getUserGuesses(uniqueId);
+            assertTrue(false);
+        }
+        catch(IllegalStateException e){
+            assertTrue(true);
+        }
+    }
```

Comment:
Sorry to insist with that, but if you modify one function you should do the same for the others!

Reply:
done